File: src/main/webapp/resources/js/controllers/NomenclatureController.js
/**
 * Created by Ruslan on 12/28/2016.
 */
angular.module('nomenclature_app', [])
    .controller('nomenclature_controller', ['$scope', '$http',
            function ($scope, $http) {

                $scope.fetchNomenclatureList = function () {
                    $http.get('nomenclature/nomenclatureList.json').success(function (nomenclatures) {
                        $scope.nomenclatures = nomenclatures;
                    });
                };

                $scope.addNomenclature = function (nomenclatureName) {
                    $http({
                        method: 'POST',
                        url: 'nomenclature/addNomenclature',
                        data: {
                            id: 0,
                            name: nomenclatureName
                        }
                    }).then(function (response) {
                        $scope.fetchNomenclatureList();
                        $scope.nomenclatureName = '';
                        $scope.message = '';
                    },function (response) {
                            $scope.message = response.data;
                    });
                };

                $scope.removeAllNomenclatures = function () {
                    $http.get('nomenclature/removeAllNomenclatures').success(function () {
                        $scope.fetchNomenclatureList();
                    });
                };

                $scope.removeNomenclature = function (nomenclature) {
                    $http({
                        method: 'POST',
                        url: 'nomenclature/removeNomenclature',
                        data: nomenclature
                    }).then(function () {
                        $scope.fetchNomenclatureList();
                    });
                };

                $scope.fetchNomenclatureList();
            }
        ]
    );

File: src/main/java/com/ruslan/pricelist/controller/NomenclatureController.java
package com.ruslan.pricelist.controller;

import com.ruslan.pricelist.beans.Nomenclature;
import com.ruslan.pricelist.service.NomenclatureService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.io.IOException;
import java.util.List;

/**
 * Created by Ruslan on 12/28/2016.
 */
@Controller
@RequestMapping(value = "/nomenclature")
public class NomenclatureController {
    @Autowired
    NomenclatureService nomenService;

    @RequestMapping("/nomenclatureList.json")
    public @ResponseBody List<Nomenclature> getNomenclatureList() throws IOException {
        return nomenService.getNomenclatures();
    }

    @RequestMapping(value = "/addNomenclature", method=RequestMethod.POST)
    public @ResponseBody
    ResponseEntity addNomenclature(@RequestBody Nomenclature nomenclature){
        try{
            nomenService.addNomenclature(nomenclature);
        }catch (IllegalArgumentException ex){
            return new ResponseEntity<>(ex.getMessage(),HttpStatus.BAD_REQUEST);
        }
        return new ResponseEntity<>("Номенклатура успешно добавлено!",HttpStatus.OK);
    }

    @RequestMapping("/removeNomenclature")
    public @ResponseBody void removeNomenclature (@RequestBody Nomenclature nomenclature){
        nomenService.removeNomenclature(nomenclature);
    }

    @RequestMapping("/removeAllNomenclatures")
    public @ResponseBody void removeAllNomenclatures(){
        nomenService.removeAllNomenclatures();
    }

    @RequestMapping
    public String getNomenclaturePage(){
        return "index";
    }
}
